### ISSUE TITLE
Vector tiles missing from Klokan CDN as well as in extracts

### ISSUE THREAD

**Issue by stephankn** [osm2vectortiles/osm2vectortiles#116]
Some vectortiles are completely missing and return 404

Example returning 404:
http://klokantech-3.tileserver.com/v1/14/12767/7560.pbf?key=WQ5sRntXphOrWgesmtmU

This is also happening with country extracts, so fault must happened in the planet already.

I inspected my extract file *thailand.mbtiles* and noticed that in the tiles/map table zoom_level=14 tile_column=12767 tile_row=8823 is missing.

The 7560 is flipped by *tileserver.php* and results in 2^14-1-7560 being 8823. Nearby tiles in the same row are available. In the column 12767 the rows 8822 to 8824 are missing.

More investigation turned out that a lot more columns have data missing. Based on the extract I expect 706 columns. Often I get less

This query for reproduction

```sql
SELECT  tile_column, rows FROM (
    SELECT tile_column, count() AS rows FROM map WHERE zoom_level=14 GROUP BY tile_column 
) WHERE rows<>706
````

returns this for *thailand.mbtiles*

tile_column | rows
------------|-----
12620 | 704
12709 | 704
12720 | 701
12721 | 701
12722 | 699
12723 | 699
12724 | 701
12740 | 705
12741 | 705
12757 | 704
12766 | 705
12767 | 703
12782 | 704
12905 | 704
12906 | 702
12964 | 705
12965 | 705
12970 | 704
13000 | 704

In this extract 55 tiles are missing,

**Comment by lukasmartinelli** [comment]
Thanks for the detailed inspection and report!

It affects distinct tiles and not a entire subpyramid - so it is not the "remove subpyramids" process.
On the top of my head I think certain tiles **failed** in Mapnik when rendering the planet.
I need to dig through the Mapnik logs from CloudWatch to search for those tiles to see a error message that Mapnik spit out when rendering those tiles. Or rerender parts where we know tiles are missing.

I thought I ran my `verify.py` script to check whether each of the subpyramids that were merged together contains all tiles- but apparently not. We need to introduce a better verification process in the next rerendering then.

**Comment by klokan** [comment]
- Is there a problem with the sample tile: zoom_level=14 tile_column=12767 tile_row=8823 during generating?

**Comment by stephankn** [comment]
Is it possible to replicate the generation process on my side? Are all required scripts/input files available? If so I can test to create tiles here to verify. Is this the process you also followed internally? http://osm2vectortiles.org/docs/own-vector-tiles/

**Comment by lukasmartinelli** [comment]
> Is this the process you also followed internally? http://osm2vectortiles.org/docs/own-vector-tiles/

This is exactly what we do - just on a distributed scale.

**Comment by stephankn** [comment]
I did run it on my machine. Your documentation misses to mention that you need to adjust the bounding box. My first try does have no gaps. Missing lowzoom data, but maybe this is related to not touching min/max zoom settings. So can't say why your setup misses tiles.

**Comment by lukasmartinelli** [comment]
> Your documentation misses to mention that you need to adjust the bounding box.

We'll add that. Or you can create a PR as the documentation is based on GitHub pages.
https://github.com/osm2vectortiles/osm2vectortiles/tree/gh-pages/docs


> My first try does have no gaps. Missing lowzoom data, but maybe this is related to not touching min/max zoom settings. So can't say why your setup misses tiles.

Good to know. This indicates it might be some kind of race condition.
What could be is that the PostgreSQL server fails to respond in a certain time frame and Mapnik aborts the rendering of the tile - that would explain the missing tiles.

It is possible that this happended in our distributed rendering process (where the load on the different PostgreSQL servers is not always constant) while when you are doing this locally it does not occur.

We will dig into this bug once we are preparing for the next rerendering and rework the rendering scripts.
With the new update process in place #129 we have a method in the future to rerender missing or buggy tiles.

**Comment by hyperknot** [comment]
Two more missing tiles for reference:
14/13386/7149
14/13386/7148

**Comment by lukasmartinelli** [comment]
Will be fixed in #111.
